content/information-distribution/node.md

```diff
@@ -54,14 +57,104 @@ the Matrix room to join.
+`_matrix._tcp` sub-domain of the server's name which, in the example above,
+would look like `_matrix._tcp.example.com`. If such a record doesn't exist,
+client implementations **should** consider that the node is reachable using its
+server name as its FQDN, and the default Matrix encrypted federation port,
+`8448`, as its port.
+
+Most Matrix homeservers can also be reached through the standard `443` HTTPS
+port. Whether there's a Matrix `SRV` DNS record for a node or not, client
+implementations **should** try reaching the said node using the `443` port
+before trying reaching it using any other port. This means a node's address and
+port resolution **should** follow the following workflow:
+
+1. Try retrieving a Matrix `SRV` DNS record from the node's server name.
+2. If the DNS record exists, try reaching the node using the FQDN extracted from
+the record and the `443` port.
+3. If step 2 succeeds, consider the FQDN extracted from the DNS record and the
+`443` port as the final parameters to reach the node (i.e. consider the node
+unreachable if those parameters don't work).
+4. If step 2 fails, consider the FQDN and port extracted from the DNS record as
+the final parameters to reach the node (i.e. consider the node unreachable if
+those parameters don't work).
+5. If the DNS record doesn't exist, try reaching the node using its server name
+as its FQDN and the `443` port.
+6. If step 5 succeeds, consider the node's server name and the `443` port as the
+final parameters to reach the node (i.e. consider the node unreachable if those
+parameters don't work).
+7. If step 5 fails, consider the node's server name and the `8448` port as the
+final parameters to reach the node (i.e. consider the node unreachable if those
+parameters don't work).
+
+This should result in making the network traffic generated by Informo client
+implementations more difficult to detect and filter on.
+
+If a client implementation's maintainer(s) decide not to use port 443, their
+client implementation **must** implement the following workflow:
+
+1. Try retrieving a Matrix `SRV` DNS record from the node's server name.
+2. If the DNS record exists, consider the FQDN and port extracted from the
+DNS record as the final parameters to reach the node (i.e. consider the node
+unreachable if those parameters don't work).
+3. If the DNS record doesn't exist, consider the node's server name and the
+`8448` port as the final parameters to reach the node (i.e. consider the node
+unreachable if those parameters don't work).
+
+{{% notice tip %}}
+Web-based client implementations **can** use the [DNS over
+HTTPS](https://tools.ietf.org/html/rfc8484) standard to perform DNS queries over
+HTTPS instead of standard DNS queries which they might not be able to perform.
 {{% /notice %}}
+
+In the event of an entry node that's currently being used becoming unreachable,
+clients implementations **must** use the generated map to provide its user(s)
```

Comment:
> I'd rather say should instead of must. Currently there are no ways to know if a server from the aliases list is harvesting data / phishing clients / filtering events / doing evil things / etc. Client administrators could instead provide a dynamic list of "approved" entry nodes.

Right, you've got a point here.

> Also "the generated map" refers to a map defined in another section. Should be something like "the map generated from the room aliases list"

Indeed.

> In the same topic, I was thinking that it could be nice to let TAs trust / blacklist some entry nodes the same way they can trust / blacklist other TAs and sources, in order to curate the raw alias-generated list

That could be a great idea. It doesn't belong to this SCS , though.